shelter/utils/manageprotocol.go

```diff
@@ -0,0 +1,72 @@
+package utils
+
+import (
+        "fmt"
+        "encoding/json"
+)
+
+const (
+        MRENCLAVE_HASH_SIZE = 32
+        MRSINGER_HASH_SIZE = 32
+        MANAGE_POOL_MAXSIZE = 2048
+        ManageCmd1 = "GETENCLAVEINFO"
+)
+
+var ReceiveMsg [MANAGE_POOL_MAXSIZE]byte
+var ReceiveMsgLen uint = 0
+
+type GetEnclaveInfo struct{
+        MsgType string          `json:"string_type"`
+}
+
+type EnclaveInfo struct{
+        Id string               `json:"id,omitempty"`
+        Message string          `json:"message,omitempty"`
+        Mrenclave string        `json:"mrenclave,omitempty"`
+        Mrsigner  string        `json:"mrsigner,omitempty"`
+        Version uint8           `json:"version,omitempty"`
+}
+
+func ConstructSendmsg(cmd string)([]byte, uint, error) {
+        cmdstr := cmd
+        var cmdmsg GetEnclaveInfo
+        switch cmdstr {
+        case ManageCmd1:
+                cmdmsg = GetEnclaveInfo{
+                        MsgType: "GETENCLAVEINFO",
+                }
+        default:
```

Comment:
Confirm it again, here should be `type` instead of `msgtype`.
The data sent should look like this：`{"type": "GETENCLAVEINFO"}`.

Reply:
Got it, thx

Reply:
change it in define by setting the json section as:
MsgType string `json:"type"`